core/src/main/java/com/linecorp/armeria/common/stream/NoopSubscriber.java

```diff
@@ -16,16 +16,28 @@
 
 package com.linecorp.armeria.common.stream;
 
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletableFuture;
+
 import org.reactivestreams.Subscriber;
 import org.reactivestreams.Subscription;
 
+import com.linecorp.armeria.internal.common.stream.StreamMessageUtil;
+
 /**
  * A {@link Subscriber} that discards all elements silently.
  */
 public final class NoopSubscriber<T> implements Subscriber<T> {
 
     private static final NoopSubscriber<?> INSTANCE = new NoopSubscriber<>();
 
+    private CompletableFuture<Void> whenSubscribed = new CompletableFuture<>();
+
+    CompletableFuture<Void> whenSubscribed() {
+        return whenSubscribed;
+    }
+
```

Comment:
This doesn't seem to be necessary after all because we can just return `whenComplete()` in `subscribe()`. Please correct me if I'm wrong.

Reply:
I misunderstood 😅  I'll put it back !